cpp/src/gandiva/engine.h

```diff
@@ -93,6 +93,9 @@ class GANDIVA_EXPORT Engine {
   /// the main module.
   Status LoadPreCompiledIR();
 
+  // load external pre-compiled IR modules from LLVMIRStore
+  Status LoadExternalPreCompiledIR();
```

Comment:
This would be more flexible if it took a `LLVMIRStore` instance as argument, no?

Reply:
See the comment here https://github.com/apache/arrow/pull/38116#discussion_r1364803533

Currently, users of Gandiva don't use the `Engine` class directly, instead, they typically use `Projector` or `Filter` classes when working with expressions, and `Projector` internally uses `LLVMGenerator`, which internally uses `Engine` class. Even if we make the API here to take an `LLVMIRStore` instance, `LLVMGenerator` will still have to populate this argument with a global variable. As far as I can tell, putting the logic in `Engine` instead of `LLVMGenerator` is more consistent with current logic, currently the `LoadPreCompiledIR ` is loading IR from a static variable, which is not too different with `LoadExternalPreCompiledIR` which loads IR from some global variable.